src/v/cloud_storage/types.h

```diff
@@ -47,6 +47,11 @@ using local_segment_path
 using s3_connection_limit
   = named_type<size_t, struct archival_s3_connection_limit_t>;
 
+/// Version of the segment metadata struct
+enum class segment_meta_version : int16_t { v1 = 1, v2 = 2 };
```

Comment:
in serde IIRC all enum class values should be serialized as int32_t so i don't think you need to specify the size unless its used for some other reason.